internal/pkg/docker/dockerengine/dockerengine.go

```diff
@@ -170,12 +171,22 @@ func (c DockerCmdClient) Login(uri, username, password string) error {
 }
 
 // Push pushes the images with the specified tags and ecr repository URI, and returns the image digest on success.
-func (c DockerCmdClient) Push(ctx context.Context, uri string, w io.Writer, tags ...string) (digest string, err error) {
+func (c DockerCmdClient) Push(ctx context.Context, uri string, engine string, w io.Writer, tags ...string) (digest string, err error) {
```

Comment:
i feel like we should have a different function for podman push (same signature as `Push` it's just with a different name), so that we don't need to change the function signature and also the push function is quite different when the engine type is podman.